fusedrug/utils/cluster/cluster_using_mmseqs.py

```diff
@@ -149,24 +159,52 @@ def cluter_impl(
     print(
         "B.2 - cluster the remaining unique representatives into different clusters based on the requested sequence identity threshold"
     )
-    mmseqs_tmp_2_for_clustering = os.path.join(output_dir, "mmseqs_DB_tmp_2")
-    clustered_db = os.path.join(output_dir, "mmseqs_DB_clustered")
-    cmd = f"mmseqs {cluster_method} {step_B_initial_db} {clustered_db} {mmseqs_tmp_2_for_clustering} -c 1.0 --min-seq-id {cluster_min_seqeunce_identity} --threads 32"
+    mmseqs_tmp_2_for_clustering = join(output_dir, "mmseqs_workspace", "mmseqs_DB_tmp_2")
+    clustered_db = join(output_dir, "mmseqs_workspace", "mmseqs_DB_clustered")
+    cmd = f"mmseqs {cluster_method} {step_B_initial_db} {clustered_db} {mmseqs_tmp_2_for_clustering} -c 1.0 --min-seq-id {cluster_min_seqeunce_identity} --threads {threads}"
     _run_system_cmd(cmd)
 
     print(
         "B.3 - generate cluster TSV for convinience"
     )  # for massive datasets, we might skip this and use mmseqs output format directly (possibly worth checking if there's already a python lib that handles this)
-    clustered_tsv = os.path.join(output_dir, "clustered.tsv")
+    clustered_tsv = join(output_dir, "clustered.tsv")
     cmd = f"mmseqs createtsv {step_B_initial_db} {step_B_initial_db} {clustered_db} {clustered_tsv}"
     _run_system_cmd(cmd)
 
+    ####
```

Comment:
What is this new step? is it relevant to ab design as well?

Reply:
the point of this new step is that a FASTA file is generated that contains the clusters centers ("representatives") including their sequence

if you want to train an AB model using only the centers, it's relevant, as you can easily get them and their respective sequence in a single FASTA file